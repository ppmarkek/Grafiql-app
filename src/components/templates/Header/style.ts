import { Grid } from '@mui/material';
import { Link } from 'react-router-dom';
import { styled } from '@mui/system';

type LinkStyleProps = {
  $width?: string;
  $isSticky?: boolean;
};

export const Wrapper = styled(Grid)(({ $isSticky }: LinkStyleProps) => ({
  height: $isSticky ? '80px' : '90px',
  backgroundColor: $isSticky ? '#f0f0f0' : 'transparent',
  justifyContent: 'space-between',
  alignItems: 'center',
  padding: '0px 50px',
  position: 'sticky',
  top: '0',
  zIndex: '10',
  transition: 'all 0.5s ease-in-out',
}));

export const StyledLink = styled(Link)(({ $width }: LinkStyleProps) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  width: $width || 'auto',
  textDecoration: 'none',
  color: '#000',
  fontWeight: '700',
  fontSize: '18px',
  padding: '10px',
  borderRadius: '10px',
  transition: 'all 0.5s',

  '&:hover': {
    boxShadow:
      '0 -10px 25px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.1)',
  },
}));
