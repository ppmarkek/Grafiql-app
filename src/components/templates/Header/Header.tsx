import React, { useState, useEffect } from 'react';
import { StyledFormControl, StyledLink, Wrapper } from './style';
import {
  Grid,
  InputLabel,
  MenuItem,
  Select,
  SelectChangeEvent,
} from '@mui/material';

const Header = () => {
  const [isSticky, setIsSticky] = useState(false);
  const [language, setLanguage] = useState('');

  const handleScroll = () => {
    const offset = window.scrollY;
    setIsSticky(offset > 50);
  };

  const handleChange = (event: SelectChangeEvent) => {
    setLanguage(event.target.value as string);
  };

  useEffect(() => {
    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, []);

  return (
    <Wrapper container $isSticky={isSticky}>
      <Grid item xs={6}>
        <StyledLink $width={'150px'} to={'/'}>
          Welcome page
        </StyledLink>
      </Grid>

      <Grid container item xs={6} justifyContent={'flex-end'} gap={'20px'}>
        <StyledFormControl>
          <InputLabel id="demo-simple-select-label">Language</InputLabel>
          <Select
            labelId="demo-simple-select-label"
            id="demo-simple-select"
            value={language}
            label="Language"
            onChange={handleChange}
          >
            <MenuItem value={10}>Russian</MenuItem>
            <MenuItem value={20}>English</MenuItem>
          </Select>
        </StyledFormControl>
        <StyledLink $width={'100px'} to={'/signOut'}>
          Sign Out
        </StyledLink>
      </Grid>
    </Wrapper>
  );
};

export default Header;
