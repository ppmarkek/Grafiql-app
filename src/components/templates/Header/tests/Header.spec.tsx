import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import Header from '../Header';

describe('Header component', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('becomes sticky on scroll', () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );
    fireEvent.scroll(window, { target: { scrollY: 100 } });
    expect(container.firstChild).toHaveStyle('position: sticky');
    expect(container.firstChild).toMatchSnapshot();
  });
});
