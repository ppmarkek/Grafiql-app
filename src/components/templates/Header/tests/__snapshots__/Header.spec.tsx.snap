// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component becomes sticky on scroll 1`] = `
<div
  class="MuiGrid-root MuiGrid-container css-jt49ft-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
  >
    <a
      class="css-1avdwde"
      href="/"
      width="150px"
    >
      Welcome page
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-1wdn4sc-MuiGrid-root"
  >
    <div
      class="MuiFormControl-root css-1325rt7-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1k5x8it-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="demo-simple-select-label"
      >
        Language
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-18a5grb-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-controls=":r1:"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="demo-simple-select"
          role="combobox"
          tabindex="0"
        >
          EN
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="en"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Language
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <a
      class="css-vmh4zg"
      href="/signOut"
      width="100px"
    >
      Sign Out
    </a>
  </div>
</div>
`;

exports[`Header component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-lfpak2-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <a
        class="css-1gztb2e"
        href="/"
        width="150px"
      >
        Welcome page
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-1wdn4sc-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root css-1325rt7-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-782rkc-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          id="demo-simple-select-label"
        >
          Language
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1slsstf-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls=":r0:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="demo-simple-select"
            role="combobox"
            tabindex="0"
          >
            EN
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="en"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Language
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <a
        class="css-6xsvjt"
        href="/signOut"
        width="100px"
      >
        Sign Out
      </a>
    </div>
  </div>
</DocumentFragment>
`;
